sdks/go/node/core/parallelCaller.go

```diff
@@ -109,35 +107,34 @@ func (pc _parallelCaller) Call(
 				&callID,
 				rootCallID,
 			)
-
+			if childCtx.Err() != nil {
+				// context has been cancelled, so skip reporting results
+				return
+			}
+			childResults <- childResult{
+				callID:  childCallID,
+				err:     err,
+				outputs: outputs,
+			}
 		}(childCall)
 	}
 
-	// subscribe to events
-	// @TODO: handle err channel
-	eventChannel, _ := pc.pubSub.Subscribe(
-		// don't cancel w/ children; we need to read err msgs
-		parentCtx,
-		model.EventFilter{
-			Roots: []string{rootCallID},
-			Since: &startTime,
-		},
-	)
-
-	var isChildErred = false
-	outputs := map[string]*model.Value{}
+	outboundScope := inboundScope
 
-eventLoop:
-	for event := range eventChannel {
-		if event.CallEnded != nil {
-			if childCallIndex, isChildCallEnded := childCallIndexByID[event.CallEnded.Call.ID]; isChildCallEnded {
-				childCallOutputsByIndex[childCallIndex] = event.CallEnded.Outputs
-				if event.CallEnded.Error != nil {
-					isChildErred = true
+	for {
```

Comment:
I'm trying to understand the goals behind this set of changes. With this change, child calls must now run on the current node right? Also making this not powered off the event channel it seems like we're also losing the ability to resume/replay because all state is in process as opposed to the stateful pubsub. What if the current node dies or is killed.. If we're relying on in process state rather than the event channel then we don't have an ability to gracefully restart and resume. None of these are hard blockers but I don't necessarily understand  the goal; it seems like it moves us farther away multi-node and HA/Fault tolerance.

Reply:
> With this change, child calls must now run on the current node right?

This is unaffected. Child calls are run in the same place they currently are, in a goroutine in the current execution context - line 101 in the new code, line 103 in old.

If in future something started running child calls on a remote node, this would happen in the `caller` interface (`caller.Call`). That 

> Also making this not powered off the event channel it seems like we're also losing the ability to resume/replay because all state is in process as opposed to the stateful pubsub.

The events being emitted are the same. The change for me is that it's now more clear that the caller is in charge of actually managing the call behavior, whether it's a kill, replay, restart or whatever, not the orchestrators as well (parallelCaller, etc) (they all delegate to the caller). It's much easier to manage state if the node dies or is killed, since it's all based off a single context object which is explicitly controlled by the caller, instead of being managed in multiple places. If in future calls are started on remote nodes, there's a single place for that logic to go - `caller.Call`.

Reply:
It's true that the current code calls caller directly, but the outputs are event driven which is where I'd propose we want to be. There's been an iterative ongoing effort to make the call graph supportive of multi-node. Some details still TBD, but In the case of recursive call types like parallel calls or serial calls, the idea was they wouldn't call caller in process, they'd fire some event like CallRequested and wait on some event like CallCompleted. This would allow those calls to be processed by some other node in entirety which has interesting characteristics like load balancing, the current recursive call being frozen and resumed on some other node, etc...

Reply:
I still don't think this prevents us from moving in that direction. This definitely makes the call graph simpler and better tested. (https://github.com/opctl/opctl/pull/906#discussion_r606947366 is a good example of where it's easier to actually return panics as errors, instead of just logging)

Reply:
Nothing prevents anything ever; all changes are always just a PR away : ). All I'm saying is IMHO this undoes some work we've done to try to move us in that direction; specifically the getting child call results via events as opposed to expecting the child calls ran in our local golang process.